File: leetcode/C++/maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int t=0,sol=0,idx=0;
        vector<int> visit(10001,0);
        for(int i=0; i<nums.size(); ++i){
            if(++visit[nums[i]]>1){
                for(; idx<nums.size(); ++idx){
                    --visit[nums[idx]]; t-=nums[idx];
                    if(nums[idx]==nums[i]) {++idx; break;}
                }
            }
            t+=nums[i];
            sol=max(sol,t);
        }
        return sol;
    }
};

File: leetcode/Python/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        visit, sol, t, idx = [0 for i in range(10001)], 0, 0, 0
        for i, v in enumerate(nums):
            if visit[v] > 0:
                while idx < i:
                    t-=nums[idx]
                    visit[nums[idx]]-=1
                    if v == nums[idx]:
                        idx+=1
                        break
                    idx+=1
            visit[v]+=1
            t+=v
            sol=max(sol,t)
        return sol